sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -2225,6 +2246,154 @@ object Radar {
         })
     }
 
+    /**
+     * Sends a custom event.
+     *
+     * @see [](https://radar.io/documentation/api#sen-event)
+     *
+     * @param[name] The name of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[callback] A callback.
+     */
+    @JvmStatic
+    fun sendEvent(name: String, metadata: JSONObject?, callback: RadarSendEventCallback) {
+        if (!initialized) {
+            callback.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        trackOnce(object : RadarTrackCallback {
+            override fun onComplete(status: RadarStatus, location: Location?, events: Array<RadarEvent>?, user: RadarUser?) {
+                if (status != RadarStatus.SUCCESS || location == null) {
+                    handler.post {
+                        callback.onComplete(status)
+                    }
+
+                    return
+                }
+
+                apiClient.sendEvent(name, metadata, user, object : RadarApiClient.RadarSendEventApiCallback {
+                    override fun onComplete(status: RadarStatus, res: JSONObject?, event: RadarEvent?) {
+                        if (status != RadarStatus.SUCCESS) {
+                            handler.post {
+                                callback.onComplete(status)
+                            }
+
+                            return
+                        }
+
+                        val finalEvents: MutableList<RadarEvent>
+
+                        if (events != null && event != null) {
+                            finalEvents = events.toMutableList()
+                            finalEvents.add(0, event)
+                        } else {
+                            finalEvents = mutableListOf(event!!)
```

Comment:
for my own knowledge, was it not compiling as I was missing these `null` checks?

Reply:
Yeah.